tools/syz-declextract/run.go

```diff
@@ -72,15 +112,61 @@ func worker(outputs chan output, files chan string, binary, compilationDatabase
 
 		cmd := exec.Command(binary, "-p", compilationDatabase, file)
 		stdout, err := cmd.Output()
-		var stderr []byte
+		var stderr string
 		if err != nil {
 			var error *exec.ExitError
 			if errors.As(err, &error) {
-				stderr = error.Stderr
+				stderr = string(error.Stderr)
 			} else {
-				stderr = []byte(err.Error())
+				stderr = err.Error()
+			}
+		}
+		outputs <- output{string(stdout), stderr}
+	}
+}
+
+func renameSyscall(kernelDir string) func(string) string {
+	var rename = make(map[string]string)
+	var isValidSyscall = make(map[string]bool)
+	for _, arch := range targets.List[targets.Linux] {
+		filepath.Walk(kernelDir+arch.KernelHeaderArch, func(path string, info fs.FileInfo, err error) error {
+			if strings.HasSuffix(path, ".tbl") {
+				fi, err := os.Lstat(path)
+				if err != nil {
+					tool.Fail(err)
+				}
+				if fi.Mode()&fs.ModeSymlink != 0 { // Don't follow symlinks.
+					return nil
+				}
+				f, err := os.Open(path)
+				if err != nil {
+					tool.Fail(err)
+				}
+				s := bufio.NewScanner(f)
+				for s.Scan() {
+					fields := strings.Fields(s.Text())
+					if len(fields) < 4 || fields[0] == "#" || fields[3] == "-" || strings.HasPrefix(fields[3], "compat") ||
+						fields[3][4:] == "ni_syscall" {
+						continue
+					}
+					rename[fields[3][4:]] = fields[2]
+					isValidSyscall[fields[2]] = true
+				}
+			}
+			return nil
+		})
+	}
+
+	return func(desc string) string {
+		if desc != "" {
+			toReplace := strings.Split(desc, "$")[0]
+			if isValidSyscall[toReplace] { // Do not replace if the syscall is valid.
```

Comment:
This looks fishy. For what cases is this needed?

Reply:
```
getegid16 getegid
getegid getegid32
```
This case for example, where `getegid16` maps to `getegid` which in turn maps to `getegid32`. In this case `getegid` is valid, and there is no reason to replace it.

Reply:
> which in turn maps to

Should this mapping be applied transitively?
It maps function name from SYSCALL_DEFINE to __NR_ const, right? If we already mapped function to const, we shouldn't use const to lookup in the map that has function names as keys.

Do I understand you example correctly that we have:
```
SYSCALL_DEFINE(getegid16, int16, a)
SYSCALL_DEFINE(getegid, int32, a)
```

then in *.tbl file:
```
common __NT_getegid getegid16
common __NT_getegid32 getegid
```
?

You say we shouldn't change getegid, but then we will generate:
```
getegid$auto(int32)
```
right?

But shouldn't we generate the following for this?
```
getegid$auto(int16)
getegid32$auto(int32)
```
Note: getegid syscall has a different arg type (does not matter now for this concrete case as we generate intptr everywhere, but is still wrong and will matter in future).

Reply:
Done! All valid names should be present now!